api/routes.go

```diff
@@ -0,0 +1,14 @@
+package api
+
+// buildHTTPRoutes registers all HTTP routes and their handlers.
+func (api *API) buildHTTPRoutes() {
+	// TODO Rate-limit these from the start to prevent brute force and other forms of abuse.
+	api.Router.POST("/login", api.userLoginHandler)
+	api.Router.POST("/user", api.userHandlerPOST)
+	api.Router.GET("/user/", api.userHandlerGET)
```

Comment:
Should this be `/user/:id` ?

Reply:
I'm not sure about these routes at all.

Yes, probably `/user/:id` would be the right thing to do and then limit that to be accessed only by the user with that id. For that we need some kind of session code first. I'll change it to have `:id` and I'll leave it open for the moment with a TODO to limit it.